src/server/services/import/import.js

```diff
@@ -81,6 +110,54 @@ const importOtherSlug = async (data, { slug, user, idField }) => {
   };
 };
 
+const assertRequiredFields = (data, requiredFields) => {
+  const keys = Object.keys(data);
+  if (requiredFields.some((key) => !keys.includes(key))) {
+    throw new Error('Missing required field');
+  }
+};
+
+const createOrUpdateCandidate = async (data, { slug: relationName }) => {
+  assertRequiredFields(data, ['firstName', 'lastName', 'party', 'email', 'published']);
+
+  const publishedAt = data.published === 'true' ? new Date() : undefined;
+
+  const where = { email: data.email };
+
+  let [candidate] = await strapi.db.query(relationName).findMany({ where });
+
+  if (!candidate) {
+    candidate = await strapi.db.query(relationName).create({ data: { ...data, publishedAt } });
```

Comment:
If possible, it would be nice to count both created and updated candidates and nominations and show the numbers at the end of the process.